### DISCUSSION TITLE
Getting this exception "Request 'ping' is not supported.". SFTP client

### DISCUSSION THREAD

**Discussion by lampedampe** [sshnet/SSH.NET#1218] [Q&A]
As title suggests, i'm getting this error : "Request '{0}' is not supported.", especially when dealing with large batches or large files. 

It's been suggested in the team that the server does not accept ICMP, and that the SSH.NET attempts to send a 'Ping' via ICMP. However when looking at the code i can't see that it's being used, and it sound strange that you would, so i doubt that.

Has anyone experienced the same issue, and has an idea of how to solve it?

**Comment by lampedampe** [comment]
So it's the server that sends ping@openssh.com message as a channel request. According to the documentation below this is not the expected behaviour, since it should be sent in the transport layer. Could there be an issue in the clients implementation which picks up this message as a channel request, or is it the server's implementation?

https://cvsweb.openbsd.org/src/usr.bin/ssh/PROTOCOL?rev=HEAD

**Comment by mus65** [comment]
@lampedampe we are actually currently experiencing a similiar issue with a ProFTPd Server:

```
System.NotSupportedException: Request 'keepalive@proftpd.org' is not supported.
   at Renci.SshNet.SubsystemSession.WaitOnHandle(WaitHandle waitHandle, Int32 millisecondsTimeout)
   at Renci.SshNet.Sftp.SftpSession.RequestRealPath(String path, Boolean nullOnError)
   at Renci.SshNet.Sftp.SftpSession.GetCanonicalPath(String path)
   at Renci.SshNet.SftpClient.InternalUploadFile(Stream input, String path, Flags flags, SftpUploadAsyncResult asyncResult, Action`1 uploadCallback)
```

There is a documentation here from ProFTPd http://www.proftpd.org/docs/howto/KeepAlives.html (see "How SSH KeepAlive Works
"). From my understanding, the client is supposed to respond to unknown messages and SSH.NET fails with this exception instead. Looking at the code, I managed to work around this by configuring a keep-alive response for this particular message:

```
connectionInfo.ChannelRequests.Add("keepalive@proftpd.org", new KeepAliveRequestInfo());
```

This seems to work.

I imagine setting a client `KeepAliveInterval` lower than the server interval may also work, but I haven't tested this.

**Comment by mus65** [comment]
This should be fixed by #1363 .
